spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -67,17 +80,22 @@
 		this.credentialsProvider = (gcpFirestoreProperties.getCredentials().hasKey()
 				? new DefaultCredentialsProvider(gcpFirestoreProperties)
 				: credentialsProvider);
+
+		this.hostPort = gcpFirestoreProperties.getHostPort();
+		this.firestoreRootPath = String.format(ROOT_PATH_FORMAT, this.projectId);
```

Comment:
Can `firestoreRootPath` optionally be overwritten by `GcpFirestoreProperties` or it doesn't make sense?
Maybe all you need the user to be able to overwrite is the `(default)` part.

Reply:
In the client libs for `FirestoreImpl` you can see that the root path is inferred based on the machine's environment: https://github.com/googleapis/google-cloud-java/blob/61a759a4cea68b5db662dbbcce4447824d53483a/google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java#L87

(they build the path from the projectId and databaseId which is always `(default)`).

So as discussed offline, it's like a "prefix" as you mentioned. We could make the prefix configurable, but I hesitate because it's hard to get it right, you have to figure out that the databaseId is *always* `(default)` which requires some digging... So I prefer just to let users do everything relative to the root to keep it simple.